SAMLPortal/Controllers/SetupController.cs

```diff
@@ -0,0 +1,141 @@
+using System.Linq;
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using SAMLPortal.Misc;
+using SAMLPortal.Models;
+using SAMLPortal.Models.Setup;
+using MySql.Data.MySqlClient;
+using Microsoft.EntityFrameworkCore;
+using CountryData.Standard;
+
+namespace SAMLPortal.Controllers
+{
+	[Route("Setup")]
+	[ServiceFilter(typeof(SetupAsyncActionFilter))]
+	public class SetupController : Controller
+	{
+		[HttpGet]
+		[Route("0")]
+		[Route("")]
+		public IActionResult Index()
+		{
+			return View();
+		}
+
+		[HttpGet]
+		[Route("1")]
+		public IActionResult FirstStep()
+		{
+			FirstStepModel model = new FirstStepModel();
+			model.MySQLPort = 3306;
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("1")]
+		public IActionResult FirstStep(FirstStepModel model)
+		{
+			if (ModelState.IsValid)
+			{
+				try
+				{
+					string connectionString = "server=" + model.MySQLHost
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'model' before using it.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)